### PR TITLE
chore(ada): add pledge type

### PR DESCRIPTION
**Pull request by theMobiusStrip**
EA-420

### CODE REVIEW

**Review by sachushaji — COMMENTED** [review]
Seems there are commitlint errors

File: modules/sdk-coin-ada/test/unit/stakingPledgeBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Transaction, TransactionBuilderFactory } from '../../src';
+import { coins } from '@bitgo/statics';
+import { rawTx } from '../resources';
+import should from 'should';
+import { TransactionType } from '@bitgo/sdk-core';
+
+describe('ADA Staking Pledge Transaction Builder', async () => {
+  const factory = new TransactionBuilderFactory(coins.get('tada'));
+
+  it('should read a prebuilt transaction', async () => {
```
Comment (ewangbitgo) [inline modules/sdk-coin-ada/test/unit/stakingPledgeBuilder.ts:10]: can we add a test to make sure rebuild txn is the same as raw data

**Review by islamaminBitGo — DISMISSED** [review]
sdk-core changes LGTM.

File: modules/sdk-coin-ada/src/lib/transaction.ts [hunk 2]
```diff
@@ -217,14 +217,21 @@ export class Transaction extends BaseTransaction {
...
-            this._type = TransactionType.StakingActivate;
-          }
-          if (cert.as_stake_deregistration() !== undefined) {
-            this._type = TransactionType.StakingDeactivate;
-          }
+          certs.push(cert);
+        }
+
+        if (
+          certs.some((c) => c.as_stake_delegation() !== undefined) &&
```
[the remark this answers is not in the document]
Comment (evanzbitgo) [reply modules/sdk-coin-ada/src/lib/transaction.ts:227]: Yeah, we will have to refine the type derivation logic later. Currently we cannot differentiate Pledge and Activate. We need to come back to check if the delegation address is the same as pool cert's owner address

**Review by ewangbitgo — APPROVED** [review]

**Review by margueriteblair — APPROVED** [review]

**Pull request merged**